R/container.R

```diff
@@ -0,0 +1,174 @@
+#' Declare post-processing for model predictions
+#'
+#' @param mode The model's mode, one of `"unknown"`, `"classification"`, or
+#' `"regression"`. Modes of `"censored regression"` are not currently supported.
+#' @param type The model sub-type. Possible values are `"unknown"`, `"regression"`,
+#' `"binary"`, or `"multiclass"`.
+#' @param outcome The name of the outcome variable.
+#' @param estimate The name of the point estimate (e.g. predicted class)
+#' @param probabilities The names of class probability estimates (if any). For
+#' classification, these should be given in the order of the factor levels of
+#' the `estimate`.
+#' @param time The name of the predicted event time. (not yet supported)
+#' @param call The call to be displayed in warnings or errors.
+#' @examples
+#'
+#' container()
+#' @export
+container <- function(mode = "unknown", type = "unknown", outcome = character(0),
+                      estimate = character(0), probabilities = character(0),
+                      time = character(0), call = rlang::current_env()) {
+  dat <-
```

Comment:
Another open question is: when do we have the users specify the relevant names of the expected columns? Within a workflow, we will set these but there are a few differences between recipes and what we are doing here are: 
 
- recipe columns could be anything or type of number. For predictions, we basically know how many things should be there and, for tidymodels, we know the names.  
 
- We really don’t need the column names until fit-time. The arguments are here because we’ll need to keep them in the container at some point. 

- The `fit` method is where these values are currently handled.

Reply:
related: https://github.com/tidymodels/container/pull/1#discussion_r1578449894